### PR TITLE
Separate the initialization logs for each resource

### PR DESCRIPTION
**Pull request by oomichi**
To investigate issues easily when facing it, this commit separates
the log for each resource, services and endpoints.

fixes: #196

### CODE REVIEW

File: pkg/dns/dns.go [hunk 1]
```diff
@@ -166,11 +166,16 @@ func (kd *KubeDNS) waitForResourceSyncedOrDie() {
...
 		case <-ticker.C:
-			if kd.endpointsController.HasSynced() && kd.serviceController.HasSynced() {
-				glog.V(0).Infof("Initialized services and endpoints from apiserver")
-				return
+			if !kd.endpointsController.HasSynced() {
+				glog.V(0).Infof("Waiting for endpoints to be initialized from apiserver...")
+				continue
+			}
+			if !kd.serviceController.HasSynced() {
+				glog.V(0).Infof("Waiting for services to be initialized from apiserver...")
```
Comment (MrHohn) [inline pkg/dns/dns.go]: It is also confusing if both endpoints and services fail to be synced while this only says waiting for endpoint. Could you make it check both `endpointsController` and `serviceController` regardless?

Comment (oomichi) [reply pkg/dns/dns.go]: @MrHohn Thanks for your review, I got it.
I will update this PR soon based on your comment.

File: pkg/dns/dns.go [hunk 2]
```diff
@@ -166,11 +166,22 @@ func (kd *KubeDNS) waitForResourceSyncedOrDie() {
...
+				glog.V(0).Infof("Waiting for endpoints to be initialized from apiserver...")
+				continue
 			}
-			glog.V(0).Infof("Waiting for services and endpoints to be initialized from apiserver...")
+			if !HasSyncedServices {
+				glog.V(0).Infof("Waiting for services to be initialized from apiserver...")
+				continue
+			}
+			glog.V(0).Infof("Initialized services and endpoints from apiserver")
+			return
```
Comment (MrHohn) [inline pkg/dns/dns.go]: This is a bit odd, what about something like this?
```golang
			unsyncedResources := []string{}
			if !kd.endpointsController.HasSynced() {
				unsyncedResources = append(unsyncedResources, "endpoints")
			}
			if !kd.serviceController.HasSynced() {
				unsyncedResources = append(unsyncedResources, "services")
			}
			if len(unsyncedResources) > 0 {
				glog.V(0).Infof("Waiting for %v to be initialized from apiserver...", unsyncedResources)
				continue
			}
			glog.V(0).Infof("Initialized services and endpoints from apiserver")
			return
```

**Comment by MrHohn** [comment]
FYI travis is currently broken: https://github.com/kubernetes/dns/issues/203.

**Comment by oomichi** [comment]
@MrHohn Thanks for your review and advice.
The issue you pointed has been solved, so I will update this pr soon.

**Pull request merged**